### ISSUE TITLE
Migration changes all my columns to lower case

### ISSUE THREAD

**Issue by buzzy** [SanderGi/sqlite-auto-migrator#5]
This one had me scratching my head for a long time! It seems that the migration changes my column names, which made code fail.

Having a .sql like this:

![image](https://github.com/user-attachments/assets/ced782d7-6de0-4882-bff8-6c5eb8c0109b)

Created a column with the name "pushtoken". Notice the small "t":

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/10434072-7b7a-4b34-9102-27df58a3a345">

This makes code like this fail:

![image](https://github.com/user-attachments/assets/d4661bfc-e9a2-44ef-91c4-a1de5a7ed810)

In the map-function, the row.pushToken is undefined, as the column returned from sqlite is now row.pushtoken

I know sqlite itself is case-insensitive, but TypeScript is and it would be nice to not be forced to use all lower-case names.

**Comment by SanderGi** [comment]
Good catch. A quick workaround until I get a fix out would be to specify the desired output case in your sql queries, e.g. `SELECT id as id, pushToken as pushToken FROM sessions WHERE user = :user AND pushToken IS NOT NULL;`. I could see this being good practice in general (though especially when dealing with a case insensitive language inside a sensitive one) since the [SQLite documentation](https://www.sqlite.org/c3ref/column_name.html) otherwise provides no guarantee as to the column names returned:
![image](https://github.com/user-attachments/assets/02d2c9cb-86a3-4f3f-9995-7a71b527a60a)
Without specifying the case you want in your output, your code could theoretically break between sqlite versions (unless Bun provides stronger guarantees than the SQLite docs, haven't been able to verify if they do).

My thoughts on a more convenient solution would be to add a flag to the migrator `case_sensitive_identifiers` which defaults to false. When true, it would treat case changes to identifiers (column/table/view/trigger/virtual table names) as renames and migrate the database to match the case. This ensures backwards compatibility, can be enabled at any time to update existing identifiers to match case with the schema, and avoids potential diverging database states that could occur with other solutions. Let me know what your thoughts are and if you would prefer a flag for each identifier type.

**Comment by buzzy** [comment]
I switched to all lower-case name, so I do not really need it. However, I think it still makes sense to add it, because I would assume most people would assume their definition from the .sql-file to be left unchanged. I also think it's quite unlikely that people would use AS to specify the same name again. I have never seen that being done "in the wild".

**Comment by SanderGi** [comment]
Good points. This makes me think case sensitivity should be the default behavior of the migrator to avoid confusion. The only caveat is that it would have to use the first appearance of the column name in the schema as the canonical name. If it interpreted the schema itself case sensitively, then you could have situations where you declare a trigger/foreign key/etc. with a different case version of the column/table name. It would not make sense for the migrator to treat this as a reference to a separate column/table since that is not how SQLite would interpret it if executing the schema. This itself could cause some confusion regarding which case the migrator is using. Perhaps the migrator should throw an error when parsing a schema that uses different cases of the same identifier to avoid any confusion?

**Comment by buzzy** [comment]
Yes, that sounds logical. People should not mix cases for the same field anyways.

**Comment by SanderGi** [comment]
Apologies for the really long wait on this one (I've been swamped with paper deadlines and running my startup). The latest version [`1.3.0`](https://www.npmjs.com/package/sqlite-auto-migrator/v/1.3.0) should be case sensitive by default and supports disabling the case sensitivity with the  `ignoreNameCase = true` option to the migrator. Feel free to open a new issue if you run into problems with this
